### PR TITLE
Add Point fmt formatter

### PR DESCRIPTION
**Pull request by jellespijker**
This allows spdlog and/or fmt to output a single point as [X, Y]. With the inclusion of the fmt/ranges.h a container of points will be: [[X_0, Y_0], [X_1, Y_1], ..., [X_n, Y_n]] Usage as:

```c++
std::vector<Point> points {{100, 200}, {300, 400}, {500, 600}};
spdlog:debug("points: {}", points};
auto fmt::format("{}", points};
```

Easy debugging commence!

Created while working on: CURA-9377 💥 Slicing error with Top/Bottom line width, which I'm dropping in favor of other tickets.

Although the file currently isn't used, I still believe we should merge, because it is a helpful class to say the least.

### CODE REVIEW

**Comment by casperlamboo** [comment]
I have a slight preference for the `(X, Y)` over the `[X, Y]` notation as brackets denote ranges in mathematics while parenthesis denote tuples. Tuples describe points better in my opinion.

**Comment by jellespijker** [comment]
I'm fine with both

**Comment by rburema** [comment]
Well, if it's about points, those are (even) closer to vectors (than tuples), and then I'd prefer `<x, y>` ;-p -- At least row vectors, for _column_ vectors, you'd use matrix notation, and those _are_ square brackets.

Given that this is quite arbitrary:

However, Jelle told me in the thread on Slack that the reason was easy(er) parsing (specifically for Python). I think an actual _use_ is more important then what we'd prefer as notation.

TL;DR: Keep as is.

**Comment by rburema** [comment]
Fails for me on windows with

`C:\.conan\9e8431\1\include\fmt/core.h(2716): error C2039: 'parse': is not a member of 'fmt::v9::detail::fallback_formatter<stripped_type,char_type,void>'`

(even if I do include the IntPoint header).

**Comment by jellespijker** [comment]
Did you do a `conan install .` Looks like you don't have the used `fmt` version

**Comment by rburema** [comment]
pretty sure I did yeah
